### PR TITLE
[GPU] Support int8 dyn-quan FC

### PR DESCRIPTION
**Pull request by byungilm**
### Details:
 - Support FC dynamic quantize for 8Bit Asym model
 - Enable SLM for 8bit weight

### Tickets:
 - CVS-152990

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:62]: I gues `offset * 2` only for `COMPRESSED_WEIGHTS_INT8`

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:202]: nit: the style of indent is quite mixed in cl files. But let's not make unnecessary changes..

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:227]: Is it necessary to change this line?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:626]: could you clean up unnecessary changes?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:62]: hmm please ignore this comment at this moment.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1024]: nit: I think now we can remove type conversion here because decompression_zp is also integral type..

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1126]: nit: `seperate` -> `separate`
nit: what does `separate_wei_zp` mean?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1127]: do we need to upconvert everything to float here?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:95]: I guess `print_log` is only for debugging purpose? Do we need to have it in the function argument?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:334]: nit: `if (!is_i4_u4_u8)`?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:811]: typo here

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:1539]: `GPU_DEBUG_LOG` I guess?

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:2854]: please remove unnecessary comment

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:202]: Modified.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:227]: Fix it.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:626]: Applied

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1126]: No special meaning. any idea? I just changed it  to 'modified_calc' mean modified part in calculation for accumulated value.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1127]: At the first step of optimizing, I thought it could be overflow on half range.
But after I read this calculation, I agree it might not be necessary. 
Could I remove this upconvert with separate PR?
 I think I better not fix any calculation just before merge this PR unless this make perf regression.

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:95]: When I checked verbose=5 log, I found there were too many debug lines from this line.
Do you think we should remove this?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:334]: I added is_weight_dyn_quantizable function to make this line simple.

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:1539]: You are right.

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:2854]: removed.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:1024]: Applied

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp
Comment (isanghao) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp:106]: That is how it is supposed to work(Logging without changing code). This is basically a dead code which is not executed unless you change the code. Could you clean up in the next PR?

**Review by isanghao — APPROVED** [review]

**Pull request merged**
